docs/understanding-docker.md

```diff
@@ -17,274 +17,287 @@ weight = -90
-to hypervisor-based virtual machines. This is especially useful in high density
-environments: for example, building your own Cloud or Platform-as-a-Service. But
-it is also useful for small and medium deployments where you want to get more
-out of the resources you have.
+to hypervisor-based virtual machines, allowing you to use more of your compute
+capacity to achieve your business goals. This is useful in high density
+environments and for small and medium deployments where you need to do more with
+fewer resources.
 
 ## What is Docker's architecture?
 Docker uses a client-server architecture. The Docker *client* talks to the
 Docker *daemon*, which does the heavy lifting of building, running, and
-distributing your Docker containers. Both the Docker client and the daemon *can*
+distributing your Docker containers. The Docker client and daemon *can*
 run on the same system, or you can connect a Docker client to a remote Docker
 daemon. The Docker client and daemon communicate via sockets or through a
-RESTful API.
+REST API.
 
 ![Docker Architecture Diagram](article-img/architecture.svg)
 
 ### The Docker daemon
-As shown in the diagram above, the Docker daemon runs on a host machine. The
-user does not directly interact with the daemon, but instead through the Docker
-client.
+The Docker daemon runs on a host machine. The user uses the Docker client to
+interact with the daemon.
 
 ### The Docker client
 The Docker client, in the form of the `docker` binary, is the primary user
-interface to Docker. It accepts commands from the user and communicates back and
-forth with a Docker daemon.
+interface to Docker. It accepts commands and configuration flags from the user and
+communicates with a Docker daemon. One client can even communicate with multiple
+unrelated daemons.
 
 ### Inside Docker
-To understand Docker's internals, you need to know about three resources:
-
-* Docker images
-* Docker registries
-* Docker containers
+To understand Docker's internals, you need to know about _images_, _registries_,
+and _containers_.
 
 #### Docker images
 
-A Docker image is a read-only template. For example, an image could contain an Ubuntu
-operating system with Apache and your web application installed. Images are used to create
-Docker containers. Docker provides a simple way to build new images or update existing
-images, or you can download Docker images that other people have already created.
+A Docker _image_ is a read-only template with instructions for creating a Docker
+container. For example, an image might contain an Ubuntu operating system with
+Apache web server and your web application installed. You can build or update
+images from scratch or download and use images created by others. An image may be
+based on, or may extend, one or more other images. A docker image is described in
+text file called a _Dockerfile_, which has a simple, well-defined syntax.
+
```

Comment:
Maybe not for this PR, but going forward, may want to mention the relationship of images to services and begin to introduce "swarm mode" somewhere in this PR, because that will become a primary workflow for using Docker/DE.